dietpi/dietpi-config

```diff
@@ -534,65 +534,68 @@ A long (or insufficiently manufactured) cable may required a higher boost settin
 
 		# Native PC
 		elif (( $G_HW_MODEL == 21 )); then
-
-			local nvidia_installed=0 nvidia_text='Install'
-			dpkg-query -s nvidia-driver &> /dev/null && nvidia_installed=1 nvidia_text='[Installed] Uninstall'
-
-			local intel_installed=0 intel_text='Install'
-			dpkg-query -s i965-va-driver &> /dev/null && intel_installed=1 intel_text='[Installed] Uninstall'
-
-			local amd_installed=0 amd_text='Install'
-			dpkg-query -s xserver-xorg-video-amdgpu &> /dev/null && amd_installed=1 amd_text='[Installed] Uninstall'
-
+		
+			local gpu_current=$(sed -n '/^[[:blank:]]*CONFIG_GPU_DRIVER=/{s/^[^=]*=//p;q}' /boot/dietpi.txt)
+			gpu_current=${gpu_current,,}
+			local driver_installed=()			
+			for i in {0..3}
+			do
+				driver_installed+=('Install')
+			done
+			
+			local installed_index=-1
+			if [[ $gpu_current == 'nvidia' ]]
+			then
+				installed_index=0
+				
+			elif [[ $gpu_current == 'intel' ]]
+			then
+				installed_index=1
+				
+			elif [[ $gpu_current == 'amd' ]]
+			then
+				installed_index=2
+				
+			elif [[ $gpu_current == 'custom' ]]
+			then
+				installed_index=3
+			fi
+			
+			if (( $installed_index >= 0 ))
+			then
+				driver_installed[$installed_index]='[Installed]'	
+			fi
+						
 			G_WHIP_MENU_ARRAY=(
 
-				'Nvidia' ": $nvidia_text non-free GPU driver"
-				'Intel' ": $intel_text free VA-API GPU drivers"
-				'AMD' ": $amd_text non-free GPU driver"
+				'Nvidia' ": ${driver_installed[0]} | Proprietary GPU driver"
+				'Intel' ": ${driver_installed[1]} | Open-Source VA-API GPU drivers"
+				'AMD' ": ${driver_installed[2]} | Proprietary GPU driver"
+				'Custom' ": ${driver_installed[3]} | Manual User Install"
+				'None' ": Uninstall all GPU drivers"
 			)
 
 			G_WHIP_MENU 'Please select an option:' || { Back_or_Exit 1; return 0; } # Display Options
 
-			if [[ $G_WHIP_RETURNED_VALUE == 'Nvidia' ]]
+			local gpu_new=$gpu_current
+			if [[ $G_WHIP_RETURNED_VALUE ]]
 			then
-				if (( $nvidia_installed ))
-				then
-					G_AGP nvidia-driver nvidia-driver-libs nvidia-xconfig libgl1-nvidia-glx
-				else
-					local headers=()
-					dpkg-query -s 'linux-image-amd64' &> /dev/null && headers+=('linux-headers-amd64')
-					dpkg-query -s 'linux-image-cloud-amd64' &> /dev/null && headers+=('linux-headers-cloud-amd64')
-					dpkg-query -s 'linux-image-rt-amd64' &> /dev/null && headers+=('linux-headers-rt-amd64')
-					G_AGI "${headers[@]}" nvidia-driver
-				fi
-
-			elif [[ $G_WHIP_RETURNED_VALUE == 'Intel' ]]
-			then
-				if (( $intel_installed ))
-				then
-					G_AGP i965-va-driver xserver-xorg-video-intel
-				else
-					G_AG_CHECK_INSTALL_PREREQ i965-va-driver libgl1-mesa-dri
-				fi
-
-			elif [[ $G_WHIP_RETURNED_VALUE == 'AMD' ]]
-			then
-				if (( $amd_installed ))
-				then
-					G_AGP firmware-amd-graphics xserver-xorg-video-amdgpu
-				else
-					G_AG_CHECK_INSTALL_PREREQ firmware-amd-graphics xserver-xorg-video-amdgpu libgl1-mesa-dri
-				fi
+				gpu_new=${G_WHIP_RETURNED_VALUE,,}
 			fi
```

Comment:
`$G_WHIP_RETURNED_VALUE` is assured to be set. When hitting "Cancel" or interrupt/terminate the menu, one falls back to the parent display menu.
```suggestion
			local gpu_new=${G_WHIP_RETURNED_VALUE,,}
```